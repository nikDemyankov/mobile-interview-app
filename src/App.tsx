import React from 'react';
import 'react-native-gesture-handler';
import { NavigationContainer } from '@react-navigation/native';
import { StatusBar, LogBox } from 'react-native';
import { Provider } from 'react-redux';

import Navigation from './navigation';
import { color } from './ui/theme';
import store from './store';

// Silence require cycle for fetch.
if (__DEV__) {
  LogBox.ignoreLogs(['Require cycle: node_modules/react-native/Libraries/Network/fetch.js']);
}

const App = () => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <StatusBar barStyle="dark-content" backgroundColor={color.background} />
        <Navigation />
      </NavigationContainer>
    </Provider>
  );
};

export default App;
