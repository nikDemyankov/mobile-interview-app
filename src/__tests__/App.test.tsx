import React from 'react';
import { create } from 'react-test-renderer';
import App from '../App';

jest.mock('@react-navigation/native', () => ({
  NavigationContainer: 'NavigationContainer',
}));
jest.mock('react-native-gesture-handler', () => '');
jest.mock('react-native', () => ({
  StatusBar: 'StatusBar',
}));
jest.mock('react-redux', () => ({
  Provider: 'Provider',
}));
jest.mock('../navigation', () => 'Navigation');
jest.mock('../store', () => 'redux-store');

it('renders correctly', () => {
  const component = create(<App />);
  expect(component).toMatchSnapshot();
});
