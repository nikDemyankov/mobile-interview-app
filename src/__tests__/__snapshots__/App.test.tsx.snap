// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Provider
  store="redux-store"
>
  <NavigationContainer>
    <StatusBar
      backgroundColor="#FFF"
      barStyle="dark-content"
    />
    <Navigation />
  </NavigationContainer>
</Provider>
`;
